docs(resource-manifest): include oal information (#502)
citizenfx/fivem-docs: content/docs/scripting-reference/resource-manifest/resource-manifest.md

provide 'mysql-async'
```

### use_experimental_fxv2_oal

This will enable the usage of OAL (One Argument List) for Lua. This aims to correct return-types of natives and provide better performance via faster native calls.

```lua
use_experimental_fxv2_oal "yes"
```

{{% alert color="warning" %}}

> This feature is still experimental and also **requires** [Lua 5.4](#lua54) to be used.

Reviewer: ```suggestion
This feature is still experimental and **requires** [Lua 5.4](#lua54) to be used.
```

{{% /alert %}}

#### Vector unpacking

> Vector unpacking is not a feature when using OAL, this means that you will have to explicitly unpack each coordinate into the native. Example below.

Reviewer: ```suggestion
Vector unpacking does not work when using OAL, this means that you will have to manually unpack coordinates instead of providing the `vector3`.
```

```lua
local coords = vector3(1, 2, 3)

SetEntityCoords(ped, coords) -- would normally work, but NOT with OAL
SetEntityCoords(ped, coords.x, coords.y, coords.z) -- works both with OAL, and without
```

#### Downsides

> Using OAL does have its downsides, if used incorrectly. Which is especially easy to do in more unexplored/unknown areas, such as RedM.
> 
> More specifically, if a native is called with incorrect parameters. Such as unpacked vectors, `0` or `1` instead of `false` or `true`, and so on.
> 
> Therefore, please ensure the types of the native parameters if you want to use OAL.

Reviewer: ```suggestion
OALs main downside is that it cannot be used if the parameter type is wrong, as internally it will be converted to whatever the underlying type is.

This means that for natives that are undocumented or don't have the right types OAL will break in unexpected ways, or likely just not working at all.
```
